File: src/com/sky/mattca/language2/parser/Parser.java
package com.sky.mattca.language2.parser;

import com.sky.mattca.language2.tokenizer.TokenString;
import com.sky.mattca.language2.tokenizer.TokenType;

import static com.sky.mattca.language2.parser.StatementParser.*;

/**
 * Created with IntelliJ IDEA.
 * User: Matt
 * Date: 28/08/13
 * Time: 23:51
 * To change this template use File | Settings | File Templates.
 */
public class Parser {

    static {
        ParserHelper parser = ParserHelper.start()
                .matchConsume(TokenType.FUNCTION)
                .matchConsume(TokenType.OPEN_BRACKET)

    }

}

File: src/com/sky/mattca/language2/parser/Parser.kt
package com.sky.mattca.language2.tests

import com.sky.mattca.language2.tokenizer.TokenString

/**
 * Created with IntelliJ IDEA.
 * User: Matt
 * Date: 13/08/13
 * Time: 16:08
 * To change this template use File | Settings | File Templates.
 */
public class Parser {

    private fun TokenString.reverse(): TokenString {
        return this;
    }

    fun testParserFunction() {
        var ts = TokenString();
        ts.reverse();
    }

}